bot/slack_im_notifier.go

```diff
@@ -3,26 +3,51 @@ package bot
 import (
 	"fmt"
 	"log"
+	"sync"
+	"time"
 
 	"github.com/adjust/michaelbot/deploy"
 	"github.com/adjust/michaelbot/slack"
 )
 
 type SlackIMNotifier struct {
-	im    *slack.InstantMessenger
-	users *slack.TeamDirectory
+	im             *slack.InstantMessenger
+	users          *slack.TeamDirectory
+	warningTimeout time.Duration
+	timer          *time.Timer
+	mutex          sync.Mutex
 }
 
-func NewSlackIMNotifier(api *slack.WebAPI) *SlackIMNotifier {
+func NewSlackIMNotifier(api *slack.WebAPI, warningTimeout time.Duration) *SlackIMNotifier {
```

Comment:
Looks like `warningTimeout` is unused?

If so, why do the tests pass?

Reply:
indeed 🙈 . I think it's because we call `DeployCompleted` directly after start so the timer just didn't have time to trigger yet. updated the test and constructor now